File: trunk/eHour-service/src/main/java/net/rrm/ehour/mail/dao/MailLogDAO.java
/**
 * Created on Apr 7, 2007
 * Created by Thies Edeling
 * Copyright (C) 2005, 2006 te-con, All Rights Reserved.
 *
 * This Software is copyright TE-CON 2007. This Software is not open source by definition. The source of the Software is available for educational purposes.
 * TE-CON holds all the ownership rights on the Software.
 * TE-CON freely grants the right to use the Software. Any reproduction or modification of this Software, whether for commercial use or open source,
 * is subject to obtaining the prior express authorization of TE-CON.
 * thies@te-con.nl
 * TE-CON
 * Legmeerstraat 4-2h, 1058ND, AMSTERDAM, The Netherlands
 *
 */

package net.rrm.ehour.mail.dao;

import java.util.List;

import net.rrm.ehour.dao.GenericDAO;
import net.rrm.ehour.domain.MailLog;
import net.rrm.ehour.domain.MailLogAssignment;

/**
 * DAO for MailLog db operations 
 **/

public interface MailLogDAO  extends GenericDAO<MailLog, Integer>
{
	/**
	 * Get mail log on project assignment id's
	 * @param projectAssignmentId
	 * @return
	 */
	public List<MailLogAssignment> findMailLogOnAssignmentIds(Integer[] projectAssignmentId);	
}

File: eHour-persistence/src/main/scala/net/rrm/ehour/persistence/mail/dao/MailLogDao.scala
package net.rrm.ehour.persistence.mail.dao

import net.rrm.ehour.domain.MailLog
import net.rrm.ehour.persistence.dao.{AbstractGenericDaoHibernateImpl, GenericDao}
import org.springframework.stereotype.Repository
trait MailLogDao extends GenericDao[Integer, MailLog] {
  def findOnEvent(event: String): List[MailLog]
}

@Repository
class MailLogDaoHibernateImpl extends AbstractGenericDaoHibernateImpl[Integer, MailLog](classOf[MailLog]) with MailLogDao {
  import scala.collection.JavaConversions._
  override def findOnEvent(event: String):List[MailLog] = findByNamedQuery("MailLog.findOnEvent", "event", event).toList
}